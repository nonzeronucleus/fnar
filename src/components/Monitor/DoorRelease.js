import React from 'react';
import styled from 'styled-components';
import switchImageOff from '../../img/switch/switch-off.png';
import switchImageOn from '../../img/switch/switch-on.png';

const SwitchImg = styled.img`
    position: absolute;
    top:400px;
    left:400px;
    z-index:4;
`;

export default ({active}) => {
    return <SwitchImg src={active ? switchImageOn : switchImageOff} />
}
