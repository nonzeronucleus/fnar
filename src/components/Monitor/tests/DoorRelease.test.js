import React from 'react';
import DoorRelease from '../DoorRelease';
import renderer from 'react-test-renderer';

it('renders correctly when not active', () => {

    const tree = renderer
        .create(
            <DoorRelease active={false} />);
    const testInstance = tree.root;

    expect(testInstance.findByProps({src:"switch-off.png"}))
    expect(tree.toJSON()).toMatchSnapshot();
});

it('renders correctly when active', () => {

    const tree = renderer
      .create(
          <DoorRelease active={true} />);
    const testInstance = tree.root;

    expect(testInstance.findByProps({src:"switch-on.png"}))

    expect(tree.toJSON()).toMatchSnapshot();
  });
