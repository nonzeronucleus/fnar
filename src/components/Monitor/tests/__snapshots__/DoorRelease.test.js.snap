// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when active 1`] = `
<img
  className="sc-bdVaJa HzQda"
  src="switch-on.png"
/>
`;

exports[`renders correctly when not active 1`] = `
<img
  className="sc-bdVaJa HzQda"
  src="switch-off.png"
/>
`;
